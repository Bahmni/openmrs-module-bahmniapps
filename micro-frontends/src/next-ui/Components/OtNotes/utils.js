import axios from 'axios';
import { OT_NOTES_BASE_URL } from "../../constants";

export const deleteOtNote = async (id) => {
    try {
        const response = await axios.delete(OT_NOTES_BASE_URL + "/" + id, {
            headers: {"Accept": "application/json", "Content-Type": "application/json", withCredentials: true}
        });
        return response.data;

    } catch (e) {
        console.error(e);
    }
}

export const constructPayload = function (noteDate, note, noteEndDate) {
    const payload = [];
    const currentDate = new Date(noteDate);
    while (currentDate <= noteEndDate) {
        payload.push({
            noteTypeName: "OT module",
            noteDate: new Date(currentDate),
            noteText: note
        });
        currentDate.setDate(currentDate.getDate() + 1);
    }
    return payload;
};

export const saveNote = async (note, startDate, endDate) => {
    try{
        const payload = endDate != null ? constructPayload(startDate, note, endDate) : [{
            noteTypeName: "OT module",
            noteDate: startDate,
            noteText: note
        }];
        const response = await axios.post(OT_NOTES_BASE_URL, payload);
        return response.data;
    }
    catch (e) {
        console.error(e);
    }
}

export const updateNoteForADay = async (noteId, note, providerUuid) => {
    const payload = {
        noteText: note,
        providerUuid: providerUuid
    };
    try{
        const response = await axios.post(OT_NOTES_BASE_URL + "/" + noteId, payload, {
            headers: {"Accept": "application/json", "Content-Type": "application/json"}
        });
        return response.data;
    }
    catch (e) {
        console.error(e);
    }
};