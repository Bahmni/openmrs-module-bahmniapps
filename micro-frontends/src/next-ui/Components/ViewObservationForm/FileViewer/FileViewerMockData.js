export const mockInfoMap = {
  providerName: "test provider",
  encounterDateTime: "encounterDateTime",
  comment: "comment",
  fileName: "fileName",
  fileType: "image/jpeg",
  fileSource: "fileSource",
};

export const mockVideoInfoMap = {
  providerName: "test provider",
  encounterDateTime: "encounterDateTime",
  comment: "comment",
  fileName: "fileName",
  fileType: "image/jpeg",
  fileSource: "fileSource",
};

export const mockIndex = 0;

export const mockItem = {
  providers: [
    {
      name: "test provider",
    },
  ],
  encounterDateTime: 1713856719000,
  comment: "comment",
  complexData: {
    title: "fileName",
    mimeType: "image/jpeg",
  },
  value: "fileSource",
};

export const mockPdfItem = {
  providers: [
    {
      name: "test provider",
    },
  ],
  encounterDateTime: 1713856719000,
  comment: "comment",
  complexData: {
    title: "fileName",
    mimeType: "application/pdf",
  },
  value: "fileSource",
};

export const mockVideoItem = {
  providers: [
    {
      name: "test provider",
    },
  ],
  encounterDateTime: 1713856719000,
  comment: "comment",
  complexData: {
    title: "fileName",
    mimeType: "video/mp4",
  },
  value: "fileSource",
};

export const observationList = [
    {
      "encounterDateTime": 1713856719000,
      "groupMembers": [],
      "providers": [
        {
          "uuid": "c1c26908-3f10-11e4-adec-0800271c1b75",
          "name": "test provider",
          "encounterRoleUuid": "a0b03050-c99b-11e0-9572-0800200c9a66"
        }
      ],
      "isAbnormal": null,
      "duration": null,
      "type": "Complex",
      "encounterUuid": "276f3f45-2bdb-4166-b495-8e3463774034",
      "interpretation": null,
      "complexData": {
        "title": "51196-Consultation-3c01db16-e63a-4b83-9159-dc3f4e8ebb01.mp4",
        "mimeType": "video/mp4",
      },
      "conceptNameToDisplay": "Patient Video",
      "concept": {
        "uuid": "eef8e969-269d-40c4-b7c9-182da11c2653",
        "name": "Patient Video",
        "dataType": "Complex",
        "shortName": "Patient Video",
      },
      "conceptUuid": "eef8e969-269d-40c4-b7c9-182da11c2653",
      "comment": "Sample comments are added",
      "value": "51200/51196-Consultation-3c01db16-e63a-4b83-9159-dc3f4e8ebb01.mp4"
    },
    {
      "encounterDateTime": 1713856719000,
      "groupMembers": [],
      "providers": [
        {
          "uuid": "c1c26908-3f10-11e4-adec-0800271c1b75",
          "name": "test provider",
        }
      ],
      "isAbnormal": null,
      "type": "Complex",
      "interpretation": null,
      "complexData": {
        "title": "51196-Consultation-37e2a799-b44c-412f-9a16-1f04ae8f9a34.mp4",
        "mimeType": "video/mp4",
      },
      "conceptNameToDisplay": "Patient Video",
      "concept": {
        "uuid": "eef8e969-269d-40c4-b7c9-182da11c2653",
        "name": "Patient Video",
        "dataType": "Complex",
        "shortName": "Patient Video",
      },
      "conceptUuid": "eef8e969-269d-40c4-b7c9-182da11c2653",
      "comment": "Sample comments are added",
      "value": "51200/51196-Consultation-37e2a799-b44c-412f-9a16-1f04ae8f9a34.mp4"
    },
    {
      "encounterDateTime": 1713856719000,
      "groupMembers": [],
      "providers": [
        {
          "uuid": "c1c26908-3f10-11e4-adec-0800271c1b75",
          "name": "test provider",
        }
      ],
      "isAbnormal": null,
      "type": "Complex",
      "interpretation": null,
      "complexData": {
        "title": "51196-Consultation-b1eae63b-98a0-4396-80c7-0beb06f5fd00.jpeg",
        "mimeType": "image/jpeg",
      },
      "conceptNameToDisplay": "Image",
      "concept": {
        "uuid": "6ca8ddf3-4e3d-4116-bd4e-a5147e580931",
        "name": "Image",
        "dataType": "Complex",
        "shortName": "Image",
      },
      "comment": "Sample comments are added",
      "value": "51200/51196-Consultation-b1eae63b-98a0-4396-80c7-0beb06f5fd00.jpeg"
    },
    {
      "encounterDateTime": 1713856719000,
      "groupMembers": [],
      "providers": [
        {
          "uuid": "c1c26908-3f10-11e4-adec-0800271c1b75",
          "name": "test provider",
          "encounterRoleUuid": "a0b03050-c99b-11e0-9572-0800200c9a66"
        }
      ],
      "isAbnormal": null,
      "type": "Complex",
      "encounterUuid": "276f3f45-2bdb-4166-b495-8e3463774034",
      "interpretation": null,
      "complexData": {
        "title": "51196-Consultation-445dd7fc-a7c7-4e9c-8ea6-70a3f567d8d4.jpeg",
        "mimeType": "image/jpeg",
      },
      "conceptNameToDisplay": "Image",
      "concept": {
        "uuid": "6ca8ddf3-4e3d-4116-bd4e-a5147e580931",
        "name": "Image",
        "dataType": "Complex",
        "shortName": "Image",
      },
      "comment": "Sample comments are added",
      "value": "51200/51196-Consultation-445dd7fc-a7c7-4e9c-8ea6-70a3f567d8d4.jpeg"
    },
    {
      "encounterDateTime": 1713856719000,
      "groupMembers": [],
      "providers": [
        {
          "uuid": "c1c26908-3f10-11e4-adec-0800271c1b75",
          "name": "test provider",
        }
      ],
      "isAbnormal": null,
      "type": "Complex",
      "interpretation": null,
      "complexData": {
        "title": "51196-Consultation-46b1cbfc-00a7-4cbb-afdf-71ac2b040182.jpeg",
        "mimeType": "image/jpeg",
      },
      "conceptNameToDisplay": "Image",
      "concept": {
        "uuid": "6ca8ddf3-4e3d-4116-bd4e-a5147e580931",
        "name": "Image",
        "dataType": "Complex",
        "shortName": "Image",
      },
      "conceptUuid": "6ca8ddf3-4e3d-4116-bd4e-a5147e580931",
      "comment": "Sample comments are added",
      "value": "51200/51196-Consultation-46b1cbfc-00a7-4cbb-afdf-71ac2b040182.jpeg"
    },
    {
      "encounterDateTime": 1713856719000,
      "visitStartDateTime": null,
      "targetObsRelation": null,
      "groupMembers": [],
      "providers": [
        {
          "uuid": "c1c26908-3f10-11e4-adec-0800271c1b75",
          "name": "test provider",
          "encounterRoleUuid": "a0b03050-c99b-11e0-9572-0800200c9a66"
        }
      ],
      "isAbnormal": null,
      "duration": null,
      "type": "Complex",
      "encounterUuid": "276f3f45-2bdb-4166-b495-8e3463774034",
      "obsGroupUuid": null,
      "creatorName": "test provider",
      "conceptSortWeight": 1,
      "parentConceptUuid": null,
      "hiNormal": null,
      "lowNormal": null,
      "formNamespace": "Bahmni",
      "formFieldPath": "Demo Final.1/1-2",
      "interpretation": null,
      "status": "AMENDED",
      "encounterTypeName": null,
      "complexData": {
        "data": null,
        "title": "51196-Consultation-aa0a417a-0a61-4c97-b4be-fe3fc43ea371.pdf",
        "mimeType": "application/pdf",
        "length": null
      },
      "unknown": false,
      "orderUuid": null,
      "abnormal": null,
      "observationDateTime": 1713856755000,
      "conceptNameToDisplay": "Image",
      "valueAsString": "51200/51196-Consultation-aa0a417a-0a61-4c97-b4be-fe3fc43ea371.pdf",
      "concept": {
        "uuid": "6ca8ddf3-4e3d-4116-bd4e-a5147e580931",
        "name": "Image",
        "dataType": "Complex",
        "shortName": "Image",
        "units": null,
        "conceptClass": "Misc",
        "hiNormal": null,
        "lowNormal": null,
        "set": false,
        "mappings": []
      },
      "conceptUuid": "6ca8ddf3-4e3d-4116-bd4e-a5147e580931",
      "voided": false,
      "voidReason": null,
      "uuid": "8a643679-fcb2-4246-9b10-01a3d1eb4eb6",
      "comment": null,
      "value": "51200/51196-Consultation-aa0a417a-0a61-4c97-b4be-fe3fc43ea371.pdf"
    }
  ];