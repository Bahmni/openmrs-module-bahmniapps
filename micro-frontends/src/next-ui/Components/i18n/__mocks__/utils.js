export function getTranslations() {
  return import("../../../../../public/i18n/locale_en.json").then(
    (module) => module.default
  );
}

export const getLocale = () => "en";
