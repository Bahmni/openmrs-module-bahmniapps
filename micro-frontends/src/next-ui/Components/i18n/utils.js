import { LS_LANG_KEY, BASE_URL } from "../../constants";

const translationsBaseUrl = "i18n";

export function getLocale() {
  return localStorage.getItem(LS_LANG_KEY) || "en";;
}

export const getTranslations = async (locale) => {
  const fileName = `locale_${locale}.json`;
  return fetchTranslations(fileName);
};

async function fetchTranslations(fileName) {
  const url = `${BASE_URL}${translationsBaseUrl}/${fileName}`;
  const response = await fetch(url);
  return response.json();
}
