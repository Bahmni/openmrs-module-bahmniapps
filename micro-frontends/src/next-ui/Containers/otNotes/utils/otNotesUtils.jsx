import axios from 'axios';
import { OT_NOTES_BASE_URL } from "../../../constants";

export const deleteOtNote = async (id) => {
    try {
        const response = await axios.delete(OT_NOTES_BASE_URL + "/" + id, {
            headers: {"Accept": "application/json", "Content-Type": "application/json", withCredentials: true}
        });
        return response.data;

    } catch (e) {
        console.error(e);
    }
}