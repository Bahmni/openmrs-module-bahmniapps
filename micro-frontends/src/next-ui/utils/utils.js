import moment from "moment";
import { defaultDateTimeFormat } from "../constants";

export const formatDate = (value, format = defaultDateTimeFormat) => {
  return value ? moment(value).format(format) : value;
};
