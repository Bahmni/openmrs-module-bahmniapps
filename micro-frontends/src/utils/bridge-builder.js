import { react2angular } from "react2angular";
/**
 * Creates a new instance of React2AngularBridgeBuilder. This helper utility creates angular wrapper components
 * for you with sensible defaults
 * 
 * @constructor
 * @param {Object} options - The options object.
 * @param {string} options.moduleName - The name of the Angular module to add the components to
 * @param {string} options.componentPrefix - The prefix to use for Angular component names.
 */
export function React2AngularBridgeBuilder({ moduleName, componentPrefix }) {
  this.moduleName = moduleName;
  this.componentPrefix = componentPrefix;
}

// The primary 2 apis
React2AngularBridgeBuilder.prototype.createComponent = createComponent;
React2AngularBridgeBuilder.prototype.createComponentWithTranslationForwarding =
  createComponentWithTranslationForwarding;

// Additional private stuff
React2AngularBridgeBuilder.prototype.__createAngularNames =
  __createAngularNames;

/** 
 * Simple private(ish) function which creates the names required for an angular component wrapper 
 * Given a component prefix of 'mfePrefix' and a component name of 'MyComponent', this function will create
 * the following names:
 * 
 * {
 *   remoteComponentName: 'mfePrefixMyComponentRemote'
 *   componentName: 'mfePrefixMyComponent'
 *   templateNameForRemote: 'mfe-prefix-my-component-remote'
 * }
*/
function __createAngularNames(reactComponentName) {
  const base = capitaliseFirst(reactComponentName);
  const remoteComponentName = `${this.componentPrefix}${base}Remote`;

  return {
    remoteComponentName,
    componentName: `${this.componentPrefix}${base}`,
    templateNameForRemote: camelCaseToHyphenatedLowerCase(remoteComponentName),
  };
}

/**
 * 
 * Registers the given component with angular and creates an intermediatary wrapper component which forwards
 * the translation function as 'tx' to the react component.
 * 
 * Given a name of 'MyComponent' and a prefix of 'mfePrefix', this function will create the following components:
 * 1/2. mfePrefixMyComponentRemote - Core react component wrapped
 * 2/2. mfePrefixMyComponent - Angular component which forwards hostData, hostApi, and tx to the remote component
 * 
 * @param {string} name name of the component in PascalCase
 * @param {function} reactComponent the react component to wrap
 * @returns {object} { componentName: 'mfePrefixMyComponent' }
 */
function createComponentWithTranslationForwarding(name, reactComponent) {
  const { remoteComponentName, componentName, templateNameForRemote } =
    this.__createAngularNames(name);

  angular
    .module(this.moduleName)
    .component(remoteComponentName, react2angular(reactComponent));

  angular.module(this.moduleName).component(componentName, {
    controller: translationForwardingController,
    bindings: {
      hostData: "=",
      hostApi: "=",
    },
    template: `
      <${templateNameForRemote}
        host-data="hostData"
        host-api="hostApi"
        tx="tx"
        app-service="appService"
      ></${templateNameForRemote}>
    `,
  });

  return {
    componentName,
  };
}


/**
 * Registers the give react component with the angular module wrapping it with react2angular
 * 
 * Given a name of 'MyComponent' and a prefix of 'mfePrefix', this function will create the following components:
 * 1/1. mfePrefixMyComponent - Angular component which wraps the react component
 * 
 * @param {string} name name of the component
 * @param {function} reactComponent the react component to wrap
 * @returns {object} { componentName: 'mfePrefixMyComponent' }
 */
function createComponent(name, reactComponent) {
  const { componentName } = this.__createAngularNames(name);

  angular
    .module(this.moduleName)
    .component(componentName, react2angular(reactComponent));

  return { componentName };
}

/**
 * A controller which forwards hostData & hostApi, but along with that, forwards the translation
 * function as tx
 */
function translationForwardingController($scope, $translate, appService) {
  const vm = this;
  $scope.hostData = vm.hostData;
  $scope.hostApi = vm.hostApi;
  $scope.tx = $translate.instant.bind($translate);
  $scope.appService = appService;
}

translationForwardingController.$inject = ["$scope", "$translate","appService"];

/** Utilities */
function camelCaseToHyphenatedLowerCase(str) {
  return str.replace(/([a-z])([A-Z])/g, "$1-$2").toLowerCase();
}

function capitaliseFirst(str) {
  return str.charAt(0).toUpperCase() + str.slice(1);
}
