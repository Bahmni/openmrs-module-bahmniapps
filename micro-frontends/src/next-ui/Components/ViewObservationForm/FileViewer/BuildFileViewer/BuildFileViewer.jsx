import React from "react";
import propTypes from "prop-types";
import { Link } from "carbon-components-react";
import { DocumentPdf, Download, PlayFilledAlt } from "@carbon/icons-react/next";
import { formatDate } from "../../../../utils/utils";
import { FileViewerModal } from "../FileViewerModal/FileViewerModal";
import { document_images_path } from "../../../../constants";
import "../FileViewer.scss";

export const BuildFileViewer = (props) => {
  const { item, index } = props;

  const [isFileViewerModalOpen, setFileViewerModal] = React.useState(false);

  const itemInfoMap = {
    providerName: item?.providers[0]?.name || "",
    encounterDateTime: formatDate(item?.encounterDateTime),
    comment: item?.comment || "",
    fileName: item?.complexData?.title,
    fileType: item?.complexData?.mimeType,
    fileRelativePath: item?.value,
    fileSource: document_images_path + item?.value,
  };

  const closeModel = () => {
    setFileViewerModal(false);
  };

  if (itemInfoMap?.fileType == "video/mp4") {
    return (
      <>
        <div className={`video-viewer video-${index}`}>
          <div
            className="video-play-button"
            onClick={() => setFileViewerModal(!isFileViewerModalOpen)}
          >
            <img
              className="form-obs-video-image"
              src={document_images_path + getThumbnail(itemInfoMap.fileRelativePath, "jpg")}
              alt="image"
            />
            <div className="video-play-button-icon">
              <PlayFilledAlt size={22} />
            </div>
          </div>
          <span className="file-notes-section">
            {itemInfoMap?.comment}
            <br />
            <span className="provider-info">
              {itemInfoMap?.providerName}
              &nbsp;
              {itemInfoMap?.encounterDateTime}
            </span>
            <span className="download-link">
              <Link
                href={itemInfoMap?.fileSource}
                download={itemInfoMap.fileName}
              >
                <Download size={20} />
              </Link>
            </span>
          </span>
        </div>
        {isFileViewerModalOpen && (
          <FileViewerModal closeModel={closeModel} infoMap={itemInfoMap} />
        )}
      </>
    );
  } else {
    return (
      <>
        <div className={`image-viewer img-${index}`}>
          {itemInfoMap?.fileType === "application/pdf" ? (
            <Link
              className="pdf-link"
              href={itemInfoMap.fileSource}
              target="_blank"
            >
              <DocumentPdf size={93} />
            </Link>
          ) : (
            <div onClick={() => setFileViewerModal(!isFileViewerModalOpen)}>
              <img
                className="form-obs-image"
                src={document_images_path + getThumbnail(itemInfoMap.fileRelativePath)}
                alt="image"
              />
            </div>
          )}
          {
            <span className="file-notes-section">
              {itemInfoMap?.comment}
              <br />
              <span className="provider-info">
                {itemInfoMap.providerName}
                &nbsp;
                {itemInfoMap.encounterDateTime}
              </span>
            </span>
          }
        </div>
        {isFileViewerModalOpen && (
          <FileViewerModal closeModel={closeModel} infoMap={itemInfoMap} />
        )}
      </>
    );
  }
};

const getThumbnail = (src, extension = undefined) => {
  if (extension) {
    return (
      (src && src.replace(/(.*)\.(.*)$/, "$1_thumbnail." + extension)) || null
    );
  }
  return (src && src.replace(/(.*)\.(.*)$/, "$1_thumbnail.$2")) || null;
};

BuildFileViewer.propTypes = {
  item: propTypes.object,
  index: propTypes.number,
};

export default BuildFileViewer;
