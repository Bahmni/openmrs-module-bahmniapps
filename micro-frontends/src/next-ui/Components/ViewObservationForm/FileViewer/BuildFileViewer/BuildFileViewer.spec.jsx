import React from "react";
import { render, screen, within } from "@testing-library/react";
import { BuildFileViewer } from "./BuildFileViewer.jsx";
import { mockIndex, mockItem, mockPdfItem, mockVideoItem } from "../FileViewerMockData";
import { formatDate } from "../../../../utils/utils";

describe("BuildFileViewer", function () {

  it("should render BuildFileViewer with all the data with image", function () {
    const { container } = render(
      <BuildFileViewer item={mockItem} index={mockIndex} />
    );
    expect(container.querySelector(".form-obs-image")).not.toBeNull();
    expect(screen.getByAltText(/image/i)).toBeTruthy();
    expect(screen.getByText(mockItem.comment)).toBeTruthy();
    const provider_info = container.querySelector(".provider-info");
    expect(within(provider_info).getByText(/test provider/i)).toBeTruthy();
    expect(within(provider_info).getByText(new RegExp(formatDate(mockItem.encounterDateTime)))).toBeTruthy();
  });

  it("should render BuildFileViewer with all the data with video and download option", function () {
    const { container } = render(
      <BuildFileViewer item={mockVideoItem} index={mockIndex} />
    );
    expect(container.querySelector(".form-obs-video-image")).not.toBeNull();
    expect(container.querySelector(".video-play-button-icon")).not.toBeNull();
    expect(container.querySelector(".download-link")).not.toBeNull();
    expect(screen.getByAltText(/image/i)).toBeTruthy();
    expect(screen.getByText(mockVideoItem.comment)).toBeTruthy();
    const provider_info = container.querySelector(".provider-info");
    expect(within(provider_info).getByText(/test provider/i)).toBeTruthy();
    expect(within(provider_info).getByText(new RegExp(formatDate(mockItem.encounterDateTime)))).toBeTruthy();
  });


  it("should render BuildFileViewer with all the data with pdf link", function () {
    const { container } = render(
      <BuildFileViewer item={mockPdfItem} index={mockIndex} />
    );
    expect(container.querySelector(".pdf-link")).not.toBeNull();
    expect(screen.getByText(mockPdfItem.comment)).toBeTruthy();
    const provider_info = container.querySelector(".provider-info");
    expect(within(provider_info).getByText(/test provider/i)).toBeTruthy();
    expect(within(provider_info).getByText(new RegExp(formatDate(mockItem.encounterDateTime)))).toBeTruthy();
  });
});
