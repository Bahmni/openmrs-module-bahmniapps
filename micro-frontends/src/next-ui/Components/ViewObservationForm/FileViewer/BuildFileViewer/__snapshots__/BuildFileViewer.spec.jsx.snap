// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuildFileViewer should render BuildFileViewer 1`] = `
<div>
  <div
    class="image-viewer img-0"
  >
    <div>
      <img
        alt="image"
        class="form-obs-image"
        src="../../document_images/fileSource"
      />
    </div>
    <span
      class="file-notes-section"
    >
      comment
      <br />
      <span
        class="provider-info"
      >
        test provider
        Â 
        23 Apr 2024 12:48 pm
      </span>
    </span>
  </div>
</div>
`;
